nimble/host/include/host/ble_gatt.h

```diff
@@ -473,6 +473,23 @@ int ble_gattc_write_reliable(uint16_t conn_handle,
 int ble_gatts_notify_custom(uint16_t conn_handle, uint16_t att_handle,
                             struct os_mbuf *om);
 
+/**
+ * Sends multiple characteristic notifications on the specified
+ * attribute handles. This function consumes the mbuf of the
+ * notification value after sending notification.
+ *
+ * @param conn_handle       The connection over which to execute the
+ *                              procedure.
+ * @param att_handles       The list of attribute handles to which
+ *                              notifications have to be sent.
+ * @param notif_values      The list of notification value mbufs.
+ * @param num_handles        The number of handles to notify.
+ *
+ * @return                  0 on success; nonzero on failure.
+ */
+int ble_gatts_multi_notify_custom(uint16_t conn_handle, uint16_t * att_handles,
```

Comment:
This was my original implementation but in order to use this structure I had to define it in `ble_gatt.h`.

`ble_gatt.h` is inaccessible in both `ble_att_clt.c` and `ble_att_priv.h` (which contain the definition and header of the ATT API).

`ble_att.h` is also a place where I could declare something like this, but that file did not contain a single struct definition so I avoided it.

Which would be better? Including `ble_gatt.h` in `ble_att_priv.h` or using `ble_att.h` instead?

Reply:
hmm maybe have a "constructor" function in att that would allow for multiple calls to create ATT multinotif in loop?
That way you don't need to include ble_gatt.h (which is user api)